ocaml/xapi/message_forwarding.ml

```diff
@@ -2607,6 +2607,32 @@ module Forward = functor(Local: Custom_actions.CUSTOM_ACTIONS) -> struct
         (fun session_id rpc ->
            Client.Host.get_server_certificate rpc session_id host)
 
+    let install_server_certificate ~__context ~host ~certificate ~private_key ~certificate_chain =
+      let local_fn =
+        Local.Host.install_server_certificate
+          ~host
+          ~certificate
+          ~private_key
+          ~certificate_chain
+      in
+      let success () =
+        (* check that fingerprint of the installed cert matches? *)
```

Comment:
it is also worth checking we can contact the host we just installed a cert on?

Reply:
maybe that belongs to the code immediately after we `reconfigure_stunnel`

Reply:
I'll make an async call, so this won't be needed